src/features/client/components/CartComponent/CartDrawer.jsx

```diff
@@ -0,0 +1,152 @@
+import {
+  Button,
+  Text,
+  Flex,
+  Heading,
+  Image,
+  Box,
+  useDisclosure,
+  Drawer,
+  DrawerBody,
+  DrawerFooter,
+  DrawerHeader,
+  DrawerContent,
+  DrawerCloseButton,
+  Center,
+} from '@chakra-ui/react';
+import basketUrl from '../../../../assets/basket-icon.svg';
+import { useSelector, useDispatch } from 'react-redux';
+import { changeAmount } from '../../../../store/reducers/cartReducer';
+
+function CartDrawer() {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const cartItems = useSelector((store) => store.cart.list);
+  const dispatch = useDispatch();
+
+  const orderTotal = cartItems.length
+    ? cartItems
+        .reduce((acc, curr) => {
+          const currValue = +curr.recipePrice;
+          return acc + currValue * curr.quantity;
+        }, 0)
+        .toFixed(2)
+    : '0';
+
+  return (
+    <>
+      <Button
+        onClick={onOpen}
+        variant="ghost"
+        position="relative"
+        display={{ base: 'inline-block', md: 'none' }}
+      >
+        <Image src={basketUrl} boxSize="30px" />
+        <Text
+          padding="2px 8px"
+          bg="blue.400"
+          rounded="xl"
+          color="white"
+          fontSize="12px"
+          position="absolute"
+          bottom="0"
+          right="0"
+        >
+          {cartItems.length
```

Comment:
Move logic to component, pass function/value

Reply:
fixed